process_pr.py

```diff
@@ -1665,9 +1674,22 @@ def process_pr(
 
             # Check if the someone asked to trigger the tests
             if valid_commenter:
-                ok, v2, v3, v4 = check_test_cmd(first_line, repository, global_test_params)
+                ok, v2, v3, v4, v5 = check_test_cmd(first_line, repository, global_test_params)
                 if ok:
-                    test_comment = comment
+                    build_only = None
+                    if v5:
+                        if has_user_emoji(bot_cache, comment, repository, "+1", cmsbuild_user):
+                            continue
+                        if test_comment and (
+                            not has_user_emoji(
+                                bot_cache, test_comment, repository, "+1", cmsbuild_user
+                            )
+                        ):
```

Comment:
I don't quite get why the second condition is necessary

Reply:
so that if someone does a `please test` and `please build` in short time then we let bot process the `please test` first and then on next iteration i will process `please build` . If we do not do this then both `build_only` and `test_comment` will be set and bot will misbehave

Reply:
Does this actually happen? I thought by design we process one comment at a time...